Run Prettier on pre-commit instead of in Sail CI

sheepsteak opened the pull request:
Prettier is currently running in Sail CI but it's a bit pointless because it will fix files but not commit them back. Instead I've added `husky` and `lint-staged` to run and fix different file types with ESLint and Prettier on pre-commit so it shouldn't be needed anymore.

ryan-roemer commented:
A best practice is to have a CI check that fails CI if prettier _would_ change files. You can see this in action on: https://github.com/FormidableLabs/spectacle/blob/master/package.json#L32

```
"prettier-check": "npm run prettier -- --list-different",
```

`--list-different` exits non-zero if anything would have been changed.

sheepsteak commented:
Ah OK. I didn't know Prettier could do that 😄 

I've added it back and used `--check` which apparently is the same as `--list-different` but will output the errors in the log too.

I also fixed the existing ESLint and Prettier errors so everything passes.

lukejacksonn commented:
Although generally I don't use git hooks, this sounds reasonable to me. If @kiraarghy is happy (she set up the CI) then so am I.
